core/rawdb/accessors_chain.go

```diff
@@ -801,15 +801,70 @@ func WriteAncientBlocks(db ethdb.AncientWriter, blocks []*types.Block, receipts
 			if i > 0 {
 				tdSum.Add(tdSum, header.Difficulty)
 			}
-			if err := writeAncientBlock(op, block, header, stReceipts, tdSum); err != nil {
+			// TODO(GalaIO): blob is empty before cancun
+			var subBlobs types.BlobTxSidecars
+			if len(blobs) > 0 {
+				subBlobs = blobs[i]
+			}
+			if err := writeAncientBlock(op, block, header, stReceipts, tdSum, subBlobs); err != nil {
 				return err
 			}
 		}
 		return nil
 	})
 }
 
-func writeAncientBlock(op ethdb.AncientWriteOp, block *types.Block, header *types.Header, receipts []*types.ReceiptForStorage, td *big.Int) error {
+// ReadBlobsRLP retrieves all the transaction blobs belonging to a block in RLP encoding.
+func ReadBlobsRLP(db ethdb.Reader, hash common.Hash, number uint64) rlp.RawValue {
+	var data []byte
+	db.ReadAncients(func(reader ethdb.AncientReaderOp) error {
+		// Check if the data is in ancients
+		if isCanon(reader, number, hash) {
+			data, _ = reader.Ancient(ChainFreezerBlobTable, number)
+			return nil
+		}
+		// If not, try reading from leveldb
+		data, _ = db.Get(blockBlobsKey(number, hash))
```

Comment:
Why no error check?

Reply:
oh, here is a geth common method to handle db get, you can check other read functions.